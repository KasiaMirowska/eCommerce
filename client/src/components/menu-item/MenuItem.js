import React from 'react';
import {MenuItemContainer, BackgroundImageContainer, ContentContainer, TitleContainer, SubtitleContainer} from './MenuItem.styles';
//import './MenuItem.styles.scss';
import {withRouter} from 'react-router-dom';

// export default withRouter(function MenuItem({ title, img, size, linkUrl, history, match}) {
//     return (
//         //if the menu item has size it will get the size class or be ignored
//         <div className={`${size} menu-item`} onClick={() => history.push(`${match.url}${linkUrl}`)}>
//             <div className='background-image' style={{backgroundImage: `url(${img})`}} />
//             <div className='content'>
//                 <h1 className='title'>{title.toUpperCase()}</h1>
//                 <span className='subtitle'>SHOP NOW</span>
//             </div>
//         </div>

//     );
// })

export default withRouter(function MenuItem({ title, img, size, linkUrl, history, match}) {
    return (
        //if the menu item has size it will get the size class or be ignored
        <MenuItemContainer size={size} onClick={() => history.push(`${match.url}${linkUrl}`)}>
            <BackgroundImageContainer style={{backgroundImage: `url(${img})`}} />
            <ContentContainer>
                <TitleContainer>{title.toUpperCase()}</TitleContainer>
                <SubtitleContainer>SHOP NOW</SubtitleContainer>
            </ContentContainer>
        </MenuItemContainer>

    );
})