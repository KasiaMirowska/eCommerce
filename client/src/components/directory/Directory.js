import React from 'react';
//import './Directory.styles.scss';
import {DirectorySectionsContainer} from './Directory.styles';
import MenuItem from '../menu-item/MenuItem';
import {selectDirectorySections} from '../../redux/directory/directory.selectors'
import { connect } from 'react-redux';


export const Directory = (props) => {
    const sections = props.sections.map(section => (
        <MenuItem
            key={section.id}
            title={section.title}
            img={section.imageUrl}
            id={section.id}
            size={section.size}
            linkUrl={section.linkUrl}
        />
    ));

    return (
        <DirectorySectionsContainer>
            {sections}
        </DirectorySectionsContainer>
    );
}

const mapStateToProps = state => ({
    sections: selectDirectorySections(state)
})


export default connect(mapStateToProps)(Directory);