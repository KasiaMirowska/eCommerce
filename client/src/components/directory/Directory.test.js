import React from 'react';
import { shallow, mount } from 'enzyme';
import { Directory } from '../../components/Directory/Directory';
import MenuItem from '../menu-item/MenuItem';


describe('Directory', () => {
    const props = {
        sections: [{section: 1}, {section: 2}, {section: 3}]
    };

    const wrapper = shallow(<Directory {...props} />);
    
    it('renders as expected', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('renders all sections', () => {
        expect(wrapper.find(MenuItem).length).toEqual(props.sections.length);
    })
})