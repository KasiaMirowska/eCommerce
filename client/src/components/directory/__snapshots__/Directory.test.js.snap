// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory renders as expected 1`] = `ShallowWrapper {}`;
